cli/src/main/java/org/eclipse/hono/cli/app/AbstractApplicationClient.java

```diff
@@ -33,10 +34,10 @@
     protected String deviceId;
     @Value(value = "${connection.retryInterval}")
     protected int connectionRetryInterval;
-    protected ApplicationClientFactory clientFactory;
+    protected ApplicationClient<? extends MessageContext> client;
 
     @Autowired
-    public final void setApplicationClientFactory(final ApplicationClientFactory factory) {
-        this.clientFactory = Objects.requireNonNull(factory);
+    public final void setApplicationClient(final ApplicationClient<? extends MessageContext> client) {
```

Comment:
It would be good to add  _JavaDoc_ for this _public_ _setter_.

Reply:
Done.